scripts/release/release.sh

```diff
@@ -0,0 +1,132 @@
+export BUILD_SCRIPT="$RELEASE_SCRIPT_DIR/build.sh"
+source "${RELEASE_SCRIPT_DIR}/release_utils.sh"
+
+cd "${REPO_DIR}"
+
+export LAST_GIT_TAG="$(git tag |
+  grep -E '^v[0-9]' |
+  sort -V |
+  tail -1)"
+
+use_existing_tag=false
+
+# Do we already know what version we want to release?
+if [[ -z "${USE_TAG}" ]]; then
+  if [[ -z "${LAST_GIT_TAG}" ]]; then
+    err "No previous git tag found and it was not provided with USE_TAG env"
+  fi
+  VERSION="$(increment_tag_version ${LAST_GIT_TAG})"
+  log "Will create new version ${VERSION}"
+
+  if [ "$(git tag -l ${VERSION})" ]; then
+    err "ERROR! Tag for ${VERSION} already exists!"
+  fi
+
+  code="$(http_get_error_code "Accept: application/vnd.github.v3+json" \
+    "https://api.github.com/repos/google/hadoop-migration-assessment-tools/releases/tags/${VERSION}")"
+  if [ "${code}" != "404" ]; then
+    err "ERROR! Release with ${VERSION} tag version name already exists or http failed! Http code is ${code}"
+  fi
+else
+  VERSION="${USE_TAG}"
+  if [[ "${LAST_RELEASE_TAG}" ]]; then
+    LAST_GIT_TAG=${LAST_RELEASE_TAG}
+  fi
+  use_existing_tag=true
+  git checkout tags/"${VERSION}" -b "${VERSION}-release"
+fi
+
+log "Version name ${VERSION} verified"
+
+# Run build and integration tests
+cd "${SCRIPT_WORKING_DIR}"
+log "Build script : ${BUILD_SCRIPT}, currend dir $(pwd)"
+bash "${BUILD_SCRIPT}"
+
+# revert to initial state after running build script
+cd "${REPO_DIR}"
+
+# create and register tag for this release if it was not manually provided
+if ! [[ $use_existing_tag ]]; then
+  log "Create new tag"
+  git tag -a "${VERSION}" -m "${VERSION}"
+  git push "https://github.com/google/hadoop-migration-assessment-tools.git" "${VERSION}"
+fi
+
+log "Prepare release notes"
+
+output="$(http_post_check_status "200" "Accept: application/vnd.github.v3+json" \
+  "https://api.github.com/repos/google/hadoop-migration-assessment-tools/releases/generate-notes" \
+  '{"tag_name":"'${VERSION}'","previous_tag_name":"'${LAST_GIT_TAG}'"}')"
+release_body="$(jq -r '.body' <<<$output)"
+
+log "Create release"
+
+payload="$(
+  jq --null-input \
+    --arg tag "${VERSION}" \
+    --arg name "${VERSION}" \
+    --arg body "${release_body}" \
+    '{ tag_name: $tag, name: $name, body: $body, draft: false }'
+)"
+
+# Create new release as a draft
+response="$(http_post_check_status "201" "Accept: application/vnd.github.v3+json" \
+  "https://api.github.com/repos/google/hadoop-migration-assessment-tools/releases" "${payload}")"
+
+log "Append zip file to the release"
+
+# Attach release binary file to the release
+release_id="$(jq -r '.id' <<<${response})"
+curl \
+  --data-binary @${RELEASE_OUTPUT_FILE} \
+  --user "${GIT_CLIENT_ID}:${GIT_CLIENT_TOKEN}" \
+  -H "Content-Type: application/octet-stream" \
+  "https://${GIT_RELEASES_USERNAME}:${GIT_PSW}@uploads.github.com/repos/google/hadoop-migration-assessment-tools/releases/${release_id}/assets?name=hadoop-migration-assessment-hooks-${VERSION}.zip"
```

Comment:
Where is the `GIT_PSW` variable defined?

Reply:
Removed